research/schooltests/sql.txt

```diff
@@ -0,0 +1,126 @@
+Erstelle eine Übersicht mit allen Songtiteln und deren Erscheingungsdatum, sortiert nach dem Erscheinungsdatum (absteigend).
+
+SELECT Song.name, Song.erscheinungsDatum FROM Song ORDER BY Song.erscheinungsDatum DESC
+
+^(?:SELECT|ELECT|SLECT|SEECT|SELCT|SELET|SELEC)\s+(?:(?:Song|ong|Sng|Sog|Son)\.[`´'"]?(?:name|ame|nme|nae|nam)[`´'"]?\s*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))\s+.*)?\s*,\s*(?:Song|ong|Sng|Sog|Son)\.[`´'"]?(?:erscheinungsDatum|rscheinungsDatum|escheinungsDatum|ercheinungsDatum|ersheinungsDatum|ersceinungsDatum|erschinungsDatum|erschenungsDatum|erscheiungsDatum|erscheinngsDatum|erscheinugsDatum|erscheinunsDatum|erscheinungDatum|erscheinungsatum|erscheinungsDtum|erscheinungsDaum|erscheinungsDatm|erscheinungsDatu)[`´'"]?\s*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))\s+.*)?|(?:Song|ong|Sng|Sog|Son)\.[`´'"]?(?:erscheinungsDatum|rscheinungsDatum|escheinungsDatum|ercheinungsDatum|ersheinungsDatum|ersceinungsDatum|erschinungsDatum|erschenungsDatum|erscheiungsDatum|erscheinngsDatum|erscheinugsDatum|erscheinunsDatum|erscheinungDatum|erscheinungsatum|erscheinungsDtum|erscheinungsDaum|erscheinungsDatm|erscheinungsDatu)[`´'"]?\s*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))\s+.*)?\s*,\s*(?:Song|ong|Sng|Sog|Son)\.[`´'"]?(?:name|ame|nme|nae|nam)[`´'"]?\s*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))\s+.*)?)\s+(?:FROM|ROM|FOM|FRM|FRO)\s+[`´'"]*(?:Song|ong|Sng|Sog|Son)[`´'"]*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))?\s+.*)?\s+(?:ORDER|RDER|ODER|ORER|ORDR|ORDE)\s+(?:(?:SIBLINGS|IBLINGS|SBLINGS|SILINGS|SIBINGS|SIBLNGS|SIBLIGS|SIBLINS|SIBLING)\s+)?(?:BY|Y|B)\s+(?:\s*([`´'"]?(?:(?:Song|ong|Sng|Sog|Son)|(?:Song|ong|Sng|Sog|Son)|\w*?)[`´'"]?\.)?[`´'"]?(?:erscheinungsDatum|rscheinungsDatum|escheinungsDatum|ercheinungsDatum|ersheinungsDatum|ersceinungsDatum|erschinungsDatum|erschenungsDatum|erscheiungsDatum|erscheinngsDatum|erscheinugsDatum|erscheinunsDatum|erscheinungDatum|erscheinungsatum|erscheinungsDtum|erscheinungsDaum|erscheinungsDatm|erscheinungsDatu)[`´'"]?\s*\s+(?:(?:DESC|ESC|DSC|DEC|DES)|(?:DESC|absteigend)));?$
+
+-------------------------------
+
+Bestimme die Anzahl aller Playlisten. Gib einen beliebigen Alias-Namen an.
+
+SELECT COUNT(*) FROM Playlist 
+
+^(?:SELECT|ELECT|SLECT|SEECT|SELCT|SELET|SELEC)\s+(?:(?:COUNT|Anzahl)\s*\(\s*[`´'"]?(?:\*|.*)[`´'"]?\s*\)\s*\s*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))\s+.*)?)\s+(?:FROM|ROM|FOM|FRM|FRO)\s+[`´'"]*(?:Playlist|laylist|Paylist|Plylist|Plalist|Playist|Playlst|Playlit|Playlis)[`´'"]*(\s*(?:(?:ALIAS|LIAS|AIAS|ALAS|ALIS|ALIA)|(?:AS|S|A))?\s+.*)?;?$
+
+-------------------------------
+
+Bestimme die Gesamtspieldauer aller Songs gruppiert nach dem Genre. Ausgegeben werden soll dabei der Name des Genre, dessen Beschreibung und die Gesamtspieldauer aller zugeordneter Songs. Bennene die Spalte mit der Gesamtspieldauer wie folgt: "Spieldauer"
```

Comment:
Similar to the 1st task:
Not naming "group by" in the task explicitly would increase complexity level.

"Welche Gesamtspieldauer besitzt jedes Genre"

Reply:
changed to: "Bestimme die Gesamtspieldauer aller Songs pro Genre (Beispiel: Rock = 750h, Pop = 500h)."